Sources/CasePaths/EnumReflection.swift

```diff
@@ -67,89 +67,48 @@ public func extract<Root, Value>(case embed: @escaping (Value) -> Root, from roo
 /// - Parameter embed: An enum case initializer.
 /// - Returns: A function that can attempt to extract associated values from an enum.
 public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -> (Value?) {
-  guard let metadata = EnumMetadata(Root.self) else {
-    #if DEBUG
-      print(
-        "\(#function) can never extract values from \(Root.self) because \(Root.self) isn't an enum!"
-      )
-    #endif
-    return { _ in nil }
-  }
+  guard
+    let metadata = EnumMetadata(Root.self),
+    metadata.typeDescriptor.fieldDescriptor != nil
+  else { return { _ in nil } }
 
-  guard metadata.typeDescriptor.fieldDescriptor != nil else {
-    #if DEBUG
-      print(
-        "\(#function) can never extract values from \(Root.self) because the metadata for \(Root.self) is incomplete!"
-      )
-    #endif
-    return { _ in nil }
-  }
+  var cachedTag: UInt32?
+  var cachedStrategy: Strategy<Root, Value>?
```

Comment:
I had `cachedStrategy` non-optional because it was maybe ~2ns faster. 🤷

Reply:
Ha. I think it’s ok to not micro-optimize further. This spelling made a little more sense to me since `nil` means uncached.